addons/account/views/account_portal_templates.xml

```diff
@@ -12,30 +12,24 @@
         </xpath>
     </template>
 
-    <template id="portal_my_home_invoice" name="Invoices / Bills" inherit_id="portal.portal_my_home" customize_show="True" priority="30">
-        <xpath expr="//div[hasclass('o_portal_docs')]" position="before">
-            <t t-set="portal_client_category_enable" t-value="True"/>
-            <t t-set="portal_vendor_category_enable" t-value="True"/>
-        </xpath>
-        <div id="portal_client_category" position="inside">
-            <t t-call="portal.portal_docs_entry">
-                <t t-set="icon" t-value="'/account/static/src/img/Bill.svg'"/>
-                <t t-set="title">Your Invoices</t>
-                <t t-set="url" t-value="'/my/invoices?filterby=invoices'"/>
-                <t t-set="text">Follow, download or pay your invoices</t>
-                <t t-set="placeholder_count" t-value="'invoice_count'"/>
-            </t>
-        </div>
-        <div id="portal_vendor_category" position="inside">
-            <t t-call="portal.portal_docs_entry">
-                <t t-set="icon" t-value="'/account/static/src/img/Bill.svg'"/>
-                <t t-set="title">Our Invoices</t>
-                <t t-set="url" t-value="'/my/invoices?filterby=bills'"/>
-                <t t-set="text">Follow, download or pay our invoices</t>
-                <t t-set="placeholder_count" t-value="'bill_count'"/>
-            </t>
-        </div>
-    </template>
+    <data noupdate="1">
+        <record id="portal_invoice_client" model="portal.entry">
+            <field name="title">Your Invoices</field>
+            <field name="text">Follow, download or pay your invoices</field>
+            <field name="url">/my/invoices?filterby=invoices</field>
+            <field name="icon">/account/static/src/img/Bill.svg</field>
+            <field name="placeholder_count">invoice_count</field>
+            <field name="bg_color">bg-light</field>
+        </record>
```

Comment:
Good to have though of `noupdate` to not lose user customization each update. However, we actually should update automatically some of those information I think, at least the url, which is linked to a specific route definition.

This can probably be achieved, with two `<record`. If you use the same xml id, the second definiton would override the record created previously. So one record in noupdate and one record in update.

But I also wonder about the rest. Assuming an user that never modifies its portal value... should not we want him to receive all updates we do on all of these (title, text, icon, etc)? :thinking: Maybe nothing should be in `noupdate` :shrug: Or maybe only when it is changed by the user, it should be in noupdate (the editor could make this record as noupdate only upon editor).

Maybe you should discuss about that with Lionel :+1:

Reply:
You can check `_set_noupdate`.

Although to keep the `url` as updatable upon Odoo upgrade, I am not sure how it could be achieved. Maybe there is a way to force it to be updated with special xml stuff. Maybe check the documentation to see if you can find anything.

Or maybe the URL field could be filled with an identifier that would not be changed during upgrade:


Something like
```python
portal_entries = request.env['portal.entry'].search([])
values.update({
    'portal_entries': portal_entries,
    'portal_entries_url': {
        'INVOICES': '/my/invoices/...'
        'TASKS': '/my/...'
    },
})
```
then
```xml
<record id="portal_invoice_client" model="portal.entry">
    <field name="title">Your Invoices</field>
    <field name="url">INVOICES_URL</field>
    ...
</record>
```
and in rendering
```xml
<t t-foreach="portal_entries" t-as="entry">
    <t t-set="url" t-value="entry.url"/>
    <t t-if="url in portal_entries_url" t-set="url" t-value="portal_entries_url[url]"/>
    ...
</t>
```

=> That way, if the user somehow edit the URL, then it uses its own URL but if it does not change it, even if the record is not updated at upgrade, it can use the new URL as it would be defined python-side.